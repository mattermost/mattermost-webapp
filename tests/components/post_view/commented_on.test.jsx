// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import CommentedOn from 'components/post_view/commented_on/commented_on.jsx';
import CommentedOnFilesMessage from 'components/post_view/commented_on_files_message';

describe('components/post_view/CommentedOn', () => {
    const baseProps = {
        displayName: 'user_displayName',
        enablePostUsernameOverride: false,
        onCommentClick: jest.fn(),
        post: {
            id: 'post_id',
            message: 'text message',
            props: {
                from_webhook: true,
                override_username: 'override_username',
            },
        },
        actions: {
            showSearchResults: jest.fn(),
            updateSearchTerms: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<CommentedOn {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({enablePostUsernameOverride: true});
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(CommentedOnFilesMessage).exists()).toBe(false);

        const newPost = {
            id: 'post_id',
            message: '',
            file_ids: ['file_id_1', 'file_id_2'],
        };
        wrapper.setProps({post: newPost, enablePostUsernameOverride: false});
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(CommentedOnFilesMessage).exists()).toBe(true);
    });

    test('should call onCommentClick on click of text message', () => {
        const wrapper = shallow(<CommentedOn {...baseProps}/>);

        wrapper.find('a').first().simulate('click');
        expect(baseProps.onCommentClick).toHaveBeenCalledTimes(1);
    });

    test('should call actions.updateSearchTerms and actions.showSearchResults on handleOnClick', () => {
        const wrapper = shallow(<CommentedOn {...baseProps}/>);

        wrapper.instance().handleOnClick();

        expect(baseProps.actions.updateSearchTerms).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.updateSearchTerms).toHaveBeenCalledWith(baseProps.displayName);

        expect(baseProps.actions.showSearchResults).toHaveBeenCalledTimes(1);
    });
});
