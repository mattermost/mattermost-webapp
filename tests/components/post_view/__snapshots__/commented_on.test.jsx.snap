// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/CommentedOn should match snapshot 1`] = `
<div
  className="post__link"
>
  <span>
    <FormattedMessage
      defaultMessage="Commented on {name}{apostrophe} message: "
      id="post_body.commentedOn"
      values={
        Object {
          "apostrophe": "'s",
          "name": <a
            className="theme"
            onClick={[Function]}
          >
            user_displayName
          </a>,
        }
      }
    />
    <a
      className="theme"
      onClick={[MockFunction]}
    >
      text message
    </a>
  </span>
</div>
`;

exports[`components/post_view/CommentedOn should match snapshot 2`] = `
<div
  className="post__link"
>
  <span>
    <FormattedMessage
      defaultMessage="Commented on {name}{apostrophe} message: "
      id="post_body.commentedOn"
      values={
        Object {
          "apostrophe": "'s",
          "name": <a
            className="theme"
            onClick={[Function]}
          >
            override_username
          </a>,
        }
      }
    />
    <a
      className="theme"
      onClick={[MockFunction]}
    >
      text message
    </a>
  </span>
</div>
`;

exports[`components/post_view/CommentedOn should match snapshot 3`] = `
<div
  className="post__link"
>
  <span>
    <FormattedMessage
      defaultMessage="Commented on {name}{apostrophe} message: "
      id="post_body.commentedOn"
      values={
        Object {
          "apostrophe": "'s",
          "name": <a
            className="theme"
            onClick={[Function]}
          >
            user_displayName
          </a>,
        }
      }
    />
    <a
      className="theme"
      onClick={[MockFunction]}
    >
      <Connect(CommentedOnFilesMessage)
        parentPostId="post_id"
      />
    </a>
  </span>
</div>
`;
